### DISCUSSION TITLE
JS fetching data

### DISCUSSION THREAD

**Discussion by emeczku** [supabase/supabase#28917] [Questions]
I am building an application using tools like tanstack query, supabase, nextjs14. Currently, I have a problem with downloading data using the supabase js library. Well, when I enter the website for the first time, everything loads correctly, the problem is when the user changes the tab in the browser, e.g. to google.com. Then, when returning, it does not download data or show queries in the network or devtools tab. I use {data, error} in my await supabase queries but nothing pops up.
Additionally, sometimes it is a problem to download user data using a session, even clearing cookies does not help.

**Comment by GaryAustin1** [comment]
You will probably need to provide more info.  Many things could be going wrong.
If you are seeing a supabase request hang (console.log after it does not fire) then you might have a call in your auth event handler if you are using one.  There is a warning here about that.  https://supabase.com/docs/reference/javascript/auth-onauthstatechange  See important note.

Otherwise something is going wrong  in your flow of handling the user session.  You don't say if you are using SSR or how you are handling browser/server clients if it all.

It is usually better to ask on Discord if you need help debugging your code.

**Comment by shox404** [comment]
import { createClient } from '@supabase/supabase-js';

// Replace with your Supabase URL and API key
const SUPABASE_URL = 'https://xyzcompany.supabase.co';
const SUPABASE_KEY = 'public-anon-key';

const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);
Supabase easy to use.

**Comment by emeczku** [comment]
Nextjs 14 ssr, below I am pasting supabase middleware
```
import { createServerClient, type CookieOptions } from '@supabase/ssr';
import { NextResponse, type NextRequest } from 'next/server';

export async function updateSession(request: NextRequest) {
  let response = NextResponse.next({
    request: {
      headers: request.headers
    }
  });

  const supabase = createServerClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
    {
      cookies: {
        get(name: string) {
          return request.cookies.get(name)?.value;
        },
        set(name: string, value: string, options: CookieOptions) {
          request.cookies.set({
            name,
            value,
            ...options
          });
          response = NextResponse.next({
            request: {
              headers: request.headers
            }
          });
          response.cookies.set({
            name,
            value,
            ...options
          });
        },
        remove(name: string, options: CookieOptions) {
          request.cookies.set({
            name,
            value: '',
            ...options
          });
          response = NextResponse.next({
            request: {
              headers: request.headers
            }
          });
          response.cookies.set({
            name,
            value: '',
            ...options
          });
        }
      }
    }
  );

  const {
    data: { user }
  } = await supabase.auth.getUser();

  if (!user && !request.nextUrl.pathname.startsWith('/login')) {
    const url = request.nextUrl.clone();
    url.pathname = '/login';
    return NextResponse.redirect(url);
  }

  return response;
}
```

supabase server
```
import { createServerClient, type CookieOptions } from '@supabase/ssr';
import { cookies } from 'next/headers';

export const createClient = () => {
  const cookieStore = cookies();

  return createServerClient(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
    {
      cookies: {
        get(name: string) {
          return cookieStore.get(name)?.value;
        },
        set(name: string, value: string, options: CookieOptions) {
          try {
            cookieStore.set({ name, value, ...options });
          } catch (error) {
            // The `set` method was called from a Server Component.
            // This can be ignored if you have middleware refreshing
            // user sessions.
          }
        },
        remove(name: string, options: CookieOptions) {
          try {
            cookieStore.set({ name, value: '', ...options });
          } catch (error) {
            // The `delete` method was called from a Server Component.
            // This can be ignored if you have middleware refreshing
            // user sessions.
          }
        }
      }
    }
  );
};

```


For example: user opens the app, some requests are good, change browser tab to youtube.com and immediately go back -> decided to go to another subpage of the app and needs to download remaining data, queries are not working, data status is "Fetching".
